http/obmc_shell.hpp

```diff
@@ -0,0 +1,199 @@
+            });
+    }
+
+    void doRead()
+    {
+        if (!session)
+        {
+            BMCWEB_LOG_DEBUG << "session is closed";
+            return;
+        }
+        streamFileDescriptor.async_read_some(
+            boost::asio::buffer(outputBuffer.data(), outputBuffer.size()),
+            [this, self(shared_from_this())](
+                const boost::system::error_code& ec, std::size_t bytesRead) {
+                BMCWEB_LOG_DEBUG << "Read done.  Read " << bytesRead
+                                 << " bytes";
+                if (!session)
+                {
+                    BMCWEB_LOG_DEBUG << "session is closed";
+                    return;
+                }
+                if (ec)
+                {
+                    BMCWEB_LOG_ERROR << "Couldn't read from ssh port: " << ec;
+                    session->close("Error in connecting to ssh port");
+                    return;
+                }
+                std::string_view payload(outputBuffer.data(), bytesRead);
+                session->sendBinary(payload);
+                doRead();
+            });
+    }
+
+    crow::websocket::Connection* session;
+    boost::asio::posix::stream_descriptor streamFileDescriptor;
+    bool doingWrite;
+    int ttyFileDescriptor;
+    pid_t pid;
+
+    std::array<char, 4096> outputBuffer;
+    std::string inputBuffer;
+};
+
+static std::map<crow::websocket::Connection*, std::shared_ptr<Handler>>
+    mapHandler;
+
+inline void requestRoutes(App& app)
+{
+    BMCWEB_ROUTE(app, "/bmc-console")
+        .privileges({{"ConfigureManager"}})
+        .websocket()
+        .onopen([](crow::websocket::Connection& conn,
+                   const std::shared_ptr<bmcweb::AsyncResp>&) {
+            BMCWEB_LOG_DEBUG << "Connection " << &conn << " opened";
+            if (auto it = mapHandler.find(&conn); it == mapHandler.end())
+            {
+                auto insertData =
+                    mapHandler.emplace(&conn, std::make_shared<Handler>(&conn));
+                if (std::get<bool>(insertData))
+                {
+                    std::get<0>(insertData)->second->connect();
+                }
+            }
+        })
+        .onclose(
+            [](crow::websocket::Connection& conn, const std::string& reason) {
+                BMCWEB_LOG_DEBUG << "bmc-shell console.onclose(reason = '"
+                                 << reason << "')";
+                if (auto it = mapHandler.find(&conn); it != mapHandler.end())
+                {
+                    it->second->doClose();
+                    mapHandler.erase(it);
+                }
+            })
+        .onmessage([]([[maybe_unused]] crow::websocket::Connection& conn,
+                      const std::string& data, [[maybe_unused]] bool isBinary) {
+            if (auto it = mapHandler.find(&conn); it != mapHandler.end())
+            {
+                it->second->inputBuffer += data;
+                it->second->doWrite();
+            }
```

Comment:
Add an error message here? If we have received a onmessage call then connection should definitely exist, if it doesn't maybe the connection was removed.